packages/apply/index.ts

```diff
@@ -300,12 +308,12 @@ const bumpMain = ({
     if (pkg.vfile.extname === ".json") {
       // for javascript
       // @ts-ignore TODO bumpType should be narrowed to meet ReleaseType
-      let version = semver.inc(pkg.pkg.version, bumpType);
+      let version = previewTemplate ? semver.valid(`${pkg.pkg.version}-${previewTemplate}`) : semver.inc(pkg.pkg.version, bumpType);
```

Comment:
so having `1.2.4-x` is beneficial when there is no `1.2.4` because then you can grab the latest preview package (which has its issues if you've installed a preview package before (which i mention in another comment below)). however, like you said, if someone is relying on this feature, it'll be disrupted when `1.2.4` gets published by someone else and that functionality will be broken until they rebase with the main branch.

i didn't want to rely on people needing to rebase. you offer something and it doesn't work as intended, the blame will quickly go to covector.

Reply:
> however, like you said, if someone is relying on this feature, it'll be disrupted when 1.2.4 gets published by someone else and that functionality will be broken until they rebase with the main branch.

I think I follow your thinking until "they rebase with the main branch". I think I was working with the assumption that, if you are using a preview from a branch, that the preview package would include the branch name. I definitely did not mention that in my comment though. Would that provide enough scoping to deal with this issue?

I am not necessarily advocating for either case. I don't know that there is a clear answer in my mind. If we ignore any bumps, it feels like it implies that is a prerelease for `1.2.3`, when in reality, it is a prerelease for future higher version. If someone initially puts a bump of `major`, but than modifies it to a `minor`, that will be really confusing seeing a random prerelease for `v1` if your package isn't close to a major release.

Reply:
> > however, like you said, if someone is relying on this feature, it'll be disrupted when 1.2.4 gets published by someone else and that functionality will be broken until they rebase with the main branch.
> 
> I think I follow your thinking until "they rebase with the main branch". I think I was working with the assumption that, if you are using a preview from a branch, that the preview package would include the branch name. I definitely did not mention that in my comment though. Would that provide enough scoping to deal with this issue?

I think I added `like you said` because I was going to say something else. Anyway, the rebase part is that `1.2.4` needs to not have been published in order for semver to fetch `1.2.4-abc.2` instead of `1.2.4-abc.1`. Once `1.2.4` exists, the incremental fetch with `^1.2.4-abc.0` doesn't work. So while I'm working on my own branch, the version PR might get merged and publish `1.2.4` while the package version in my branch is still `1.2.3`. In order for me to be able to utilize the incremental logic of the preview package fetching, I need to rebase so that the version of the package I'm working on will be updated to `1.2.4` and therefore my preview packages will then be `1.2.5-abc.#`. It just raises expectations and requires too much upkeep on the dev's end.

> I am not necessarily advocating for either case. I don't know that there is a clear answer in my mind. If we ignore any bumps, it feels like it implies that is a prerelease for `1.2.3`, when in reality, it is a prerelease for future higher version. If someone initially puts a bump of `major`, but than modifies it to a `minor`, that will be really confusing seeing a random prerelease for `v1` if your package isn't close to a major release.

In my opinion I think not-bumping keeps things much simpler. Like you just mentioned, if someone modifies their change file to a major bump, it's going to publish preview packages with a much higher version number and that might create conflicts too. And also, like I mentioned in the paragraph above, `1.2.4-abc` is only a prerelease of `1.2.4` while `1.2.4` isn't published. As soon as a version PR is merged and I don't rebase, I'd be publishing "prerelease" packages of a version that already exists.

Reply:
It bothers me a bit that not bumping publishes a version "backwards", but I think we have to go this direction. You are right, this is a lot of overhead and makes it much more complex. 

The "version backwards" meaning, we already published a `1.2.3` and publishing a `1.2.3-branch-name.0` would result in a a situation where `^1.2.3-branch-name.0` returns `1.2.3`.

Let's go without bumping the version, and we could always reconsider in the future.